### ISSUE TITLE
Can otvmd execute this (old) 16 bit app: iPhotoPlus?

### ISSUE THREAD

**Issue by marcorocco2102** [otya128/winevdm#1058]
Hi, for kindness: I have an old 16bit program that I would like run in windows 10-64bit. Please using your ntdvmx64 is there any way to run it?

You can download it on my dropbox at: https://www.dropbox.com/s/tssw292rh0jwoyb/IPPLUS.rar?dl=0

and the executable is \ipplus\ipplus.exe

I have install otvmd 0.8.1 (on Windows 10 64bit build 1909 + Visual Studio 2017). Apparently the program starts but when you try to load and then save an image in .jpg format (from File -> Open \ *. jpg or load a .bmp image and go to the menu File -> SaveAs -> (*. jpg) -> option or load and save a .jpg file) an error occurs. 

Could it be possible to try to fix this problem? thanks a lot

**Comment by ArguaBILL** [comment]
ok i think maybe there's something going on with either file size or jpg formatting (yes there are different ways to format a standard jpg that wouldn't've existed back in 1993) because i was able to load this once png image i saved as a jpg with ms paint (windows 11) with absolutely no problem
![image](https://user-images.githubusercontent.com/93482203/142761299-8a1a3f30-71ab-44f0-ab17-41f5b3c2fe74.png)
i then converted the original png version of the image to a bmp, also with ms paint (windows 11), and then saved that to a jpg with your application at the highest quality settings with no problem
![image](https://user-images.githubusercontent.com/93482203/142761536-df753d35-ccc5-4a84-b271-f48037d96d54.png)

**Comment by ArguaBILL** [comment]
to be fair though i am using the latest "unstable" build of otvdm, dev2194, so maybe there was also some updated code that fixed the issue

**Comment by marcorocco2102** [comment]
I use 2161, is it possible download 2194? however if you load a .jpg image and from menu File --> Save as --> (*.jpg) "option" you see the dialog "jpeg save option"

![ipperr2](https://user-images.githubusercontent.com/92343066/142776236-ecb9f3da-c208-491a-83a2-e92ac0b4c72d.jpg)

but if you clic on "OK" or "cancel"

the program strangely freeze.

Another problem is this: when trying to paste an image from the clipboard (for example it you press the "stamp" key on the keyboard, and the screenshot of al the desktop is copy on the clipboard) and when you try to paste the image of the clip do "File" --> "new" --> user defined [1920x1080], and from menu "Edit" --> "past" --> As selection

the program return this error:

![ipperr3](https://user-images.githubusercontent.com/92343066/142776453-4e282ae5-543a-4d03-ab98-0c9d36054981.jpg)

is there a way to solve?

thankss

**Comment by ArguaBILL** [comment]
i meant to say 2184 but yes there is https://ci.appveyor.com/api/buildjobs/5valuenomcki89le/artifacts/otvdm-master-2184.zip

**Comment by cracyc** [comment]
https://github.com/otya128/winevdm/pull/1059 should fix the loading issue.

>Another problem is this: when trying to paste an image from the clipboard (for example it you press the "stamp" key on the keyboard, and the screenshot of al the desktop is copy on the clipboard) and when you try to paste the image of the clip do "File" --> "new" --> user defined [1920x1080], and from menu "Edit" --> "past" --> As selection

This probably isn't fixable.  If the program can't read the bitmap on the clipboard then there's nothing we can do about it.  Copying and paste from mspaint works fine so it's not a general problem.

**Comment by ArguaBILL** [comment]
Yeah, I bet that's because this software doesn't support the PNG format (by default all copied images and screenshots are PNG format)

**Comment by ArguaBILL** [comment]
with the newest version of OTVDM, dev2189, i was able to directly import a jpg photograph taken with my fairly new Samsung phone into the software
![image](https://user-images.githubusercontent.com/93482203/142888018-798e4cf0-cd77-4491-800b-002f96bcb707.png)

**Comment by marcorocco2102** [comment]
kindly, can I download dev2189?

however with 2184 is solved the problem of freeze when clic on File-->saveAs-->[*.jpg]"option" dialog window, but for example if I try to open this .jpg image

https://www.dropbox.com/s/ix7ogx8ecarh2g6/DSC04404.JPG?dl=0

and try to save in a new .jpg file using File-->SaveAs-->[*.jpg] c:\test.jpg

strangely the final test.jpg is of 0 Kb (a empty file).

Is there a way to solve it? thanks

**Comment by ArguaBILL** [comment]
I'll link [dev2189 ](https://ci.appveyor.com/api/buildjobs/sh49pjy0e3koqyoy/artifacts/otvdm-master-2189.zip) to you, but with these images enclosed so that way you'll know how where to get them on your own (click on what I've highlighted in the images).
![image](https://user-images.githubusercontent.com/93482203/143144444-61e7efe1-ffe0-4749-afab-f7e66d903d35.png)
![image](https://user-images.githubusercontent.com/93482203/143144479-148b18ad-7c63-4292-b77e-ab69a2e434ac.png)
![image](https://user-images.githubusercontent.com/93482203/143144614-19efe070-6893-4136-965d-3b312528d9ad.png)
![image](https://user-images.githubusercontent.com/93482203/143144654-c20e02e2-614c-4731-8fea-699c3fe9029a.png)

**Comment by marcorocco2102** [comment]
thanks, dev2189 work well but if possible is there only one think I wonder:

iPPlus report an error when trying to past something from the windows clipboard (File --New-->Current clipboard ;  Edit -->past As selection)

![errclip](https://user-images.githubusercontent.com/92343066/144744481-21d55a6a-b337-4b4f-82b0-01a77d212724.jpg)

Is there a way to let the content of the notes pass through the windows 64bit system inside the tunnel created by winedvm?

thank you very much

**Comment by cracyc** [comment]
If the data on the clipboard isn't in a format that the program can read then there's nothing winevdm can do about it.  MS paint and the Windows 10 snipping tool both work fine though.

**Comment by marcorocco2102** [comment]
I understand, however using Windows 10 21H1 X86 (and all the versions X86 of Windows with the microsoft NTVDM) the copy/paste by clipboard function works in IPhotoPlus

**Comment by marcorocco2102** [comment]
sorry semms that is not possibile to download dev2189 from the link. Plese is possible to get it? thanks

**Comment by ArguaBILL** [comment]
[just get the latest version](https://ci.appveyor.com/api/buildjobs/bkisgb9gqfsof4gc/artifacts/otvdm-master-2246.zip) (The same instructions I sent earlier in this conversation are still valid for finding the latest version on your own.)

**Comment by marcorocco2102** [comment]
can I recerive the otvdm-master-2246.zip ? seems that the link over is not working. Thanks

**Comment by emendelson** [comment]
Please read the instructions in the posts above. Start with this page:

https://github.com/otya128/winevdm

Scroll down to "Download latest version (unstable)" and click the link.

The follow the examples in ArguaBILL's post from Nov 23, 2021. If you can't find the link by following the images that he displays, study the images again, and you will find the file for yourself.
